src/sql/workbench/contrib/notebook/browser/cellViews/cellToolbar.component.ts

```diff
@@ -14,13 +22,78 @@ export const CELL_TOOLBAR_SELECTOR: string = 'cell-toolbar-component';
 	templateUrl: decodeURI(require.toUrl('./cellToolbar.component.html'))
 })
 export class CellToolbarComponent {
+	@ViewChild('celltoolbar', { read: ElementRef }) private celltoolbar: ElementRef;
+
 	public buttonEdit = localize('buttonEdit', "Edit");
 	public buttonClose = localize('buttonClose', "Close");
 	public buttonAdd = localize('buttonAdd', "Add new cell");
-	public buttonMoveDown = localize('buttonMoveDown', "Move cell down");
-	public buttonMoveUp = localize('buttonMoveUp', "Move cell up");
 	public buttonDelete = localize('buttonDelete', "Delete cell");
+	public buttonMoreActions = localize('buttonMoreActions', "More actions");
+
+	@Input() public cellModel: ICellModel;
+	@Input() set model(value: NotebookModel) {
+		this._model = value;
+	}
+	private _actionBar: Taskbar;
+	private _model: NotebookModel;
+	private _editCellAction: EditCellAction;
+
+	constructor(
+		@Inject(IInstantiationService) private _instantiationService: IInstantiationService,
+		@Inject(IContextMenuService) private contextMenuService: IContextMenuService
+	) {
+	}
+
+	get model(): NotebookModel {
+		return this._model;
+	}
+
+	ngOnInit() {
+		this.initActionBar();
+	}
+
+	private initActionBar(): void {
+		let addCodeCellButton = new AddCellAction('notebook.AddCodeCell', localize('codePreview', "Code cell"), 'notebook-button masked-pseudo code');
+		addCodeCellButton.cellType = CellTypes.Code;
+
+		let addTextCellButton = new AddCellAction('notebook.AddTextCell', localize('textPreview', "Markdown cell"), 'notebook-button masked-pseudo markdown');
+		addTextCellButton.cellType = CellTypes.Markdown;
+
+		let deleteButton = this._instantiationService.createInstance(CellToolbarAction, 'notebook.deleteCell', '', 'codicon masked-icon delete', this.buttonDelete, this.cellModel);
+
+		// Todo: Get this to show the list of actions specific to code or markdown cell.
+		let moreActionsButton = new MoreActions('notebook.moreActions', this.buttonMoreActions, 'codicon masked-icon more');
+
+		// Todo: Wire up toolbarToggleEditMode
+		// Todo: Wire up toolbarUnselectActiveCell
+		this._editCellAction = this._instantiationService.createInstance(EditCellAction, 'notebook.editCell', true);
```

Comment:
I got the icon toggle working. Now I need to trigger edit mode so the markdown editor opens. Also need to get the close button to unselect the active cell. I could not find an example of this being done.

Reply:
@chlafreniere Setting the cell as active would also put it in edit mode I would expect right? 

I'd start by looking at updateActiveCell https://github.com/microsoft/azuredatastudio/blob/master/src/sql/workbench/services/notebook/browser/models/notebookModel.ts#L391

Reply:
actually, the cell model has an `isEditMode` property that needs to be set to toggle the edit mode; active cell is just indicated in the UI by the active cell highlight.